opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -1132,6 +1143,8 @@ def __init__(
         else:
             self.id_generator = id_generator
         self._resource = resource
+        if not sampler:
+            sampler = sampling._get_from_env_or_default()
```

Comment:
Doesn't this create a new object each time as opposed to one instance previously? Not a big deal but curious what was the motivation behind this change?

Reply:
Previously, it was running whenever the module was loaded which I felt was unnecessary. Plus, it made testing harder because this defaulting happened no matter what. Plus, the defaulting would produce warnings and messages that the sampler is misconfigured, when it's just in line with the new functionality.
"Couldn't recognize sampler"
"Could not convert TRACES_SAMPLER_ARG to float."